### ISSUE TITLE
Need to be able to use a segment definition from later versions

### ISSUE THREAD

**Issue by RikiMerrick** [haffonist/igamt#888]
In LRI and LOI and eDOS we are using v2.5.1, but pre-adopted a lot of extra fields to 2.5.1 segments - example OBX-26 - 30 from v2.8.2. Need to be able to work in v2.5.1 in general, but pull from later versions (If that means pre-adopting the entire segment from prior versions, then need to be able to get to those).

**Comment by RikiMerrick** [comment]
In fact we need to change a field or a dataytpe to a different version, so we want to have 2.5.1 segment but add new field or pre-adopt a datatype from a later version.

**Comment by carolinerosin** [comment]
@robnist @haffo @Jungyubw 
We probably need to talk in person about that issue, I can explain the use case and I understand Riki's issue

**Comment by haffonist** [comment]
@RikiMerrick @carolinerosin we do support pre-adoption of datatype. 

Right click "Datatypes
Add HL7 Datatype 
Select Version
Search for Datatype
Click As Flavor
Submit
Go back to segment
Click on Pencil to select a datatype flavor
Select the preAdopted datatype flavor

**Comment by carolinerosin** [comment]
It is not just pre-adopting datatypes, it is pre-adopting new fields that do not exist in 2.5.1
I noticed we can use "ADD FIELD" (at the segment level) to manually add pre-adopted fields, and then assign the right pre-adopted datatype.

I don't find how to do that at the datatype level (to add the pre-adopted components).

**Comment by robnist** [comment]
I'm confused. Why not adopt the higher level element that contains the field you want from the version of the standard you want. Maybe I'm not following correctly, but to just add fields using the ADD FIELD is dangerous. This feature was added to support building query and response messages, not for pre-adoption.

**Comment by RikiMerrick** [comment]
We can pick the later versioned segment, but then need to set all the other
fields down to 2.5.1 default...
I really wish we wouldn't do the pre-adoption thing and rather just declare
use of the later version HL7 message altogether.

Riki

Riki Merrick
Vernetzt, LLC
rikimerrick@gmail.com
cell: 916.216.0052
skype: riki.merrick

**Comment by robnist** [comment]
Agreed. When you're always doing work-arounds it is a red flag that the design of what you're doing isn't right. 

BTW, Immunization's next guides is version 2.8.2. which is the latest. I think LAB should do the same.

**Comment by RikiMerrick** [comment]
Agreed - may be make a ballot comment to that effect?

Riki

Riki Merrick
Vernetzt, LLC
rikimerrick@gmail.com
cell: 916.216.0052
skype: riki.merrick
